nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKuduProcessor.java

```diff
@@ -260,4 +266,110 @@ protected Update updateRecordToKudu(KuduTable kuduTable, Record record, List<Str
         return update;
     }
 
+    private void addPredicate(KuduScanner.KuduScannerBuilder scannerBuilder, KuduTable kuduTable, String column, Object value, String comparisonOp) {
+        ColumnSchema columnSchema = kuduTable.getSchema().getColumn(column);
+        KuduPredicate predicate = KuduPredicate.newComparisonPredicate(columnSchema, KuduPredicate.ComparisonOp.valueOf(comparisonOp), value);
+        scannerBuilder.addPredicate(predicate);
+    }
+
+    protected void scan(ProcessContext context, ProcessSession session, KuduTable kuduTable, String predicates, List<String> projectedColumnNames, ResultHandler handler) throws Exception {
+        KuduScanner.KuduScannerBuilder scannerBuilder = this.kuduClient.newScannerBuilder(kuduTable);
+        final String[] arrayPredicates = (predicates == null || predicates.isEmpty() ? new String[0] : predicates.split(","));
+
+        for(String column : arrayPredicates){
+            if (column.contains("=")) {
+                final String[] parts = column.split("=");
+                addPredicate(scannerBuilder, kuduTable, parts[0], parts[1], "EQUAL");
+            } else if(column.contains(">")) {
+                final String[] parts = column.split(">");
+                addPredicate(scannerBuilder, kuduTable, parts[0], parts[1], "GREATER");
+            } else if(column.contains("<")) {
+                final String[] parts = column.split("<");
+                addPredicate(scannerBuilder, kuduTable, parts[0], parts[1], "LESS");
+            } else if(column.contains(">=")) {
+                final String[] parts = column.split(">=");
+                addPredicate(scannerBuilder, kuduTable, parts[0], parts[1], "GREATER_EQUAL");
+            } else if(column.contains("<=")) {
+                final String[] parts = column.split("<=");
+                addPredicate(scannerBuilder, kuduTable, parts[0], parts[1], "LESS_EQUAL");
+            }
+        }
+
+        if(!projectedColumnNames.isEmpty()){
+            scannerBuilder.setProjectedColumnNames(projectedColumnNames);
+        }
+
+        KuduScanner scanner = scannerBuilder.build();
+        for (RowResult rowResult: scanner) {
+            handler.handle(rowResult);
+        }
+    }
+
+    /**
+     * Serializes a row from Kudu to a JSON document of the form:
+     *
+     * {
+     *    "rows": [
+     *      {
+     *          "columnname-1" : "value1",
+     *          "columnname-2" : "value2",
+     *          "columnname-3" : "value3",
+     *          "columnname-4" : "value4",
+     *      },
+     *      {
+     *          "columnname-1" : "value1",
+     *          "columnname-2" : "value2",
+     *          "columnname-3" : "value3",
+     *          "columnname-4" : "value4",
+     *      }
+     *    ]
+     * }
+     */
+    protected String convertToJson(RowResult row) {
```

Comment:
Not sure if JSON should be the format of choice here.
Avro seems to be a better option. If it needs to be displayed or processed as a JSON a ConvertRecord processor could help with that.

Also (even if it builds an Avro) it probably should not be in the `AbstractKuduProcessor` but int the `ScanKuduResultHandler` itself.

Reply:
I'm thinking for initial release JSON would be a better choice, as it gives users better transparency and easy for debugging. if required user can use ConvertJSONToAvro.